test/test.googleauth.ts

```diff
@@ -1134,42 +1134,51 @@ describe('googleauth', () => {
 
   it('getCredentials should get metadata from the server when running on GCE',
      async () => {
-       nockIsGCE();
-       const isGCE = await auth._checkIsGCE();
-       assert.strictEqual(true, auth.isGCE);
        const response = {
          default: {
            email: 'test-creds@test-creds.iam.gserviceaccount.com',
            private_key: null
          }
        };
-       nock.cleanAll();
-       const scope =
-           nock(host).get(svcAccountPath).reply(200, response, HEADERS);
-       const body = await auth.getCredentials();
-       assert(body);
-       assert.strictEqual(
-           body!.client_email, 'test-creds@test-creds.iam.gserviceaccount.com');
-       assert.strictEqual(body!.private_key, undefined);
-       scope.done();
-     });
-
-  it('getCredentials should error if metadata server is not reachable',
-     async () => {
-       const scopes =
-           [nockIsGCE(), nock(HOST_ADDRESS).get(svcAccountPath).reply(404)];
+       const scopes = [
+         nockIsGCE(),
+         nock(host).get(svcAccountPath).reply(200, response, HEADERS)
+       ];
+       blockGoogleApplicationCredentialEnvironmentVariable();
+       auth._fileExists = () => false;
        await auth._checkIsGCE();
        assert.strictEqual(true, auth.isGCE);
-       await assertRejects(auth.getCredentials());
+       const body = await auth.getCredentials();
+       assert.ok(body);
+       assert.strictEqual(
+           body.client_email, 'test-creds@test-creds.iam.gserviceaccount.com');
+       assert.strictEqual(body.private_key, undefined);
        scopes.forEach(s => s.done());
      });
 
+  it('getCredentials should error if metadata server is not reachable', async () => {
```

Comment:
I might move `getCredentials` into a describe block, then the test cases read a bit better:

"it should error if metadata server is not reachable"

Reply:
So personally - I try to avoid nested `describe` blocks, because it screws up indentation and makes things harder to read.  That having been said - lets chat in depth as part of a clean up PR as you mentioned above :)